libr/io/p/io_gprobe.c

```diff
@@ -47,12 +58,125 @@ enum {
 	GPROBE_ACK = 0x0c,
 	GPROBE_RESET = 0x20,
 	GPROBE_GET_DEVICE_ID = 0x30,
+	GPROBE_GET_INFORMATION = 0x40,
 	GPROBE_RAM_READ_2 = 0x52,
 	GPROBE_RAM_WRITE_2 = 0x53,
 	GPROBE_RUN_CODE_2 = 0x54,
 };
 
-static int sp_close (struct sp_port *port) {
+#if __UNIX__
+static ut8 gprobe_checksum_i2c (const ut8 *p, unsigned int size, ut8 initial) {
+	ut8 res = initial;
+	unsigned int k;
+
+	for (k = 0; k < size; ++k)
+		res ^= p[k];
+
+	return res;
+}
+
+static void gprobe_frame_i2c (RBuffer *frame) {
+	ut8 size = r_buf_size (frame) + 1;
+	ut8 header[] = {0x51, 0x80 + size + 3, 0xc2, 0x00, 0x00};
+	ut8 checksum;
+
+	r_buf_prepend_bytes (frame, &size, 1);
+	r_buf_prepend_bytes (frame, header, sizeof (header));
+
+	checksum = gprobe_checksum_i2c (r_buf_buffer (frame), r_buf_size (frame), GPROBE_I2C_ADDR);
```

Comment:
define and assign checksum in the same line

Reply:
How?

Reply:
Sure. But don't we define variables at the beginnig of a function? Calling checksum() makes only sense after prepend().